src/pages/api/secure/send-reset-pin-link.ts

```diff
@@ -0,0 +1,63 @@
+import jwt from 'jsonwebtoken'
+import { NextApiRequest, NextApiResponse } from 'next'
+
+import { withSessionRoute } from '@/ironAuth/withSessionApiRoute'
+import { appUrl } from '@/utils/constants'
+import { sendResetPinEmail } from '@/utils/email_helper'
+
+// JWT_SECRET must be set in environment variables
+const JWT_SECRET = process.env.JWT_SECRET
+const TOKEN_EXPIRY = '5m'
+
+if (!JWT_SECRET) {
+  throw new Error('JWT_SECRET environment variable is required')
+}
+
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== 'POST') {
+    return res.status(405).json({ error: 'Method not allowed' })
+  }
+
+  try {
+    const account_address = req.session.account!.address
+
+    // Create a secure JWT with expiration and single-use capability
+    const payload = {
+      type: 'reset_pin',
+      account_address,
+      iat: Math.floor(Date.now() / 1000), // issued at
+      jti: `${account_address}-${Date.now()}-${Math.random()
+        .toString(36)
+        .substr(2, 9)}`,
+    }
+
+    const resetToken = jwt.sign(payload, JWT_SECRET!, {
+      algorithm: 'HS256',
+      expiresIn: TOKEN_EXPIRY,
+    })
+
+    const resetUrl = `${appUrl}/dashboard/reset-pin?token=${resetToken}&address=${account_address}`
+
+    // Get user's notification email from the request body
+    const { email } = req.body
```

Comment:
Again, an email shouldn't be passed from the request. The one on the DB should be the single source of truth for this account.